include/megaapi.h

```diff
@@ -13002,6 +13057,64 @@ class MegaApi
          */
         MegaNodeList* search(const char* searchString, int order = ORDER_NONE);
 
+        /**
+         * @brief Search nodes containing a search string in their name
+         *
+         * The search is case-insensitive.
+         *
+         * The search will consider every accessible node for the account:
+         *  - Cloud drive
+         *  - Inbox
+         *  - Rubbish bin
+         *  - Incoming shares from other users
+         *
+         * This function allows to cancel the processing at any time by passing a MegaCancelToken and calling
+         * to MegaCancelToken::setCancelFlag(true). If a valid object is passed, it must be kept alive until
+         * this method returns.
+         *
+         * You take the ownership of the returned value.
+         *
+         * @param searchString Search string. The search is case-insensitive
+         * @param cancelToken MegaCancelToken to be able to cancel the processing at any time.
+         * @param order Order for the returned list
+         * Valid values for this parameter are:
+         * - MegaApi::ORDER_NONE = 0
+         * Undefined order
+         *
+         * - MegaApi::ORDER_DEFAULT_ASC = 1
+         * Folders first in alphabetical order, then files in the same order
+         *
+         * - MegaApi::ORDER_DEFAULT_DESC = 2
+         * Files first in reverse alphabetical order, then folders in the same order
+         *
+         * - MegaApi::ORDER_SIZE_ASC = 3
+         * Sort by size, ascending
+         *
+         * - MegaApi::ORDER_SIZE_DESC = 4
+         * Sort by size, descending
+         *
+         * - MegaApi::ORDER_CREATION_ASC = 5
+         * Sort by creation time in MEGA, ascending
+         *
+         * - MegaApi::ORDER_CREATION_DESC = 6
+         * Sort by creation time in MEGA, descending
+         *
+         * - MegaApi::ORDER_MODIFICATION_ASC = 7
+         * Sort by modification time of the original file, ascending
+         *
+         * - MegaApi::ORDER_MODIFICATION_DESC = 8
+         * Sort by modification time of the original file, descending
+         *
+         * - MegaApi::ORDER_ALPHABETICAL_ASC = 9
+         * Sort in alphabetical order, ascending
+         *
+         * - MegaApi::ORDER_ALPHABETICAL_DESC = 10
+         * Sort in alphabetical order, descending
+         *
+         * @return List of nodes that contain the desired string in their name
+         */
+        MegaNodeList* search(const char* searchString, MegaCancelToken *cancelToken, int order = ORDER_NONE);
```

Comment:
Looks great - just one thing we might want to support - if thread B calls cancel(), and thread A was just finishing search() anyway, there's a race condition as to whether the search finished naturally, or whether it noticed the flag in time, and skipped some work.   There's no way for the caller to know which of those occurred, and the caller may want to still display the results as fully valid, if the user effectively clicked a cancel button too late.    Perhaps the MegaCancelToken could also contain a flag as to whether the cancel request had any effect?     Or maybe that's too much detail for this PR - it's certainly looking good anyway.  cheers

Reply:
It's certainly another good improvement for the future, but I think the benefit in this case, the `search()` sync call, it's very little. Furthermore, it looks to me that the management of whether the call completed its work before or after the cancellation could require to be specific to each case (it depends on what the app has requested, it's not the same a search than a request to the API, ie.). So let's keep it simple by now.
Thanks for the feedback